backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -79,22 +105,126 @@ const getObjectArrayFromRules = async (teamspace, project, model, revId, rules)
 	return { container: model, _ids: Object.values(matchedMeshes) };
 };
 
-TicketGroups.updateTicketGroup = updateGroup;
+const convert3dRepoIdsToExternalIds = async (teamspace, project, objects) => {
+	const convertedObjects = await Promise.all(objects.map(async (obj) => {
+		if (!obj._ids) {
+			return obj;
+		}
+
+		const convertedObject = { ...obj };
+
+		const shared_ids = await getNodesByIds(teamspace, project, obj.container, obj._ids,
+			{ _id: 0, shared_id: 1 });
+
+		const externalIdKeys = Object.values(externalIdNamesToKeys).flat();
+		const query = { parents: { $in: shared_ids.map((s) => s.shared_id) }, 'metadata.key': { $in: externalIdKeys } };
```

Comment:
this will work, but in the principle that shared_ids can be across revisions, we should really put rev_id in this query.

Reply:
this rev_id should I get it from nodes though ? 🤔

Reply:
or yes, you can just take it from the nodes. they should be coming from the same revision